### PR TITLE
python310Packages.ilua: init at 0.2.1

### PR DESCRIPTION
**Pull request by WolfangAukang**
###### Motivation for this change

Solves #148008

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/ilua/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ lib
+, buildPythonApplication
+, fetchPypi
+, jupyter_console
+, pygments
+, termcolor
+, txzmq }:
```
Comment (fabaff) [inline pkgs/development/python-modules/ilua/default.nix:7]:
```suggestion
, txzmq
}:
```

File: pkgs/development/python-modules/ilua/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  propagatedBuildInputs = [
+    jupyter_console
+    pygments
+    termcolor
+    txzmq
+  ];
+
+  # No tests found
+  doCheck = false;
+
```
Comment (fabaff) [inline pkgs/development/python-modules/ilua/default.nix:28]: Please add `pythonImportsCheck`.

File: pkgs/development/python-modules/ilua/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    inherit pname version;
+    sha256 = "sha256-YxV6xC7GS5NXyMPRZN9YIJxamgP2etwrZUAZjk5PjtU=";
+  };
+
+  propagatedBuildInputs = [
+    jupyter_console
+    pygments
+    termcolor
+    txzmq
+  ];
```
Comment (fabaff) [inline pkgs/development/python-modules/ilua/default.nix:24]: Upstream is mentioning [`jupyter_core`](https://github.com/guysv/ilua/blob/master/setup.cfg#L38).

File: pkgs/development/python-modules/txzmq/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pyzmq
+, twisted }:
```
Comment (fabaff) [inline pkgs/development/python-modules/txzmq/default.nix:5]:
```suggestion
, twisted
}:
```

File: pkgs/development/python-modules/txzmq/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    inherit version;
+    pname = "txZMQ";
+    sha256 = "sha256-jWB9C/CcqUYAuOQvByHb5D7lOgRwGCNErHrOfljcYXc=";
+  };
+
+  propagatedBuildInputs = [
+    pyzmq
+    twisted
+  ];
+
```
Comment (fabaff) [inline pkgs/development/python-modules/txzmq/default.nix:21]: Looks that that `pytestCheckHook` could work for the tests. 

```suggestion
  checkInputs = [
    pytestCheckHook
  ];
```

File: pkgs/development/python-modules/ilua/default.nix (hunk 3 above)
Comment (WolfangAukang) [reply pkgs/development/python-modules/ilua/default.nix:27]: Done

File: pkgs/development/python-modules/ilua/default.nix (hunk 2 above)
Comment (WolfangAukang) [reply pkgs/development/python-modules/ilua/default.nix:31]: Done

File: pkgs/development/python-modules/txzmq/default.nix (hunk 5 above)
Comment (WolfangAukang) [reply pkgs/development/python-modules/txzmq/default.nix:23]: Done. It still skips 9 tests though.

File: pkgs/development/python-modules/ilua/default.nix (hunk 1 above)
Comment (WolfangAukang) [reply pkgs/development/python-modules/ilua/default.nix]: Done

File: pkgs/development/python-modules/txzmq/default.nix (hunk 4 above)
Comment (WolfangAukang) [reply pkgs/development/python-modules/txzmq/default.nix]: Done

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 156866` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.ilua</li>
    <li>python310Packages.txzmq</li>
    <li>python39Packages.ilua</li>
    <li>python39Packages.txzmq</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/156866#issuecomment-1028161853)
- [ ] all depending packages build
- [x] tested on x86_64

Looks good to me.

**Pull request merged**